runtime/lua/vim/lsp.lua

```diff
@@ -1110,6 +1111,28 @@ function lsp.buf_notify(bufnr, method, params)
   return resp
 end
 
+
+--@private
+--- Get the details of the completion item
+function lsp._resolve_completion_detail()
+  local bufnr = resolve_bufnr()
+  local completed_item_var = vim.v.completed_item
+  if
+    completed_item_var and completed_item_var.user_data and completed_item_var.user_data.lsp and
+      completed_item_var.user_data.lsp.completion_item
+   then
+    local item = completed_item_var.user_data.lsp.completion_item
+    lsp.buf_request(bufnr, "completionItem/resolve", item, function(err, _, result)
+      if err or not result then
+        return
+      end
+      if result.additionalTextEdits then
+        util.apply_text_edits(result.additionalTextEdits, bufnr)
```

Comment:
In my knowledge, the LSP clients apply `additionalTextEdits` or snippet extensions on only ` CompleteDone`.

So I think `_on_complete_done` or` _confirm_completed_item` is a better name.

For example, some LSP clients send a `completionItem / resolve` in vim's` CompleteChanged` event to display the document.

However, Currently, this function cannot be called on CompleteChanged because it will apply additionalTextEdits.

Reply:
Makes sense. will update